VogtObserver3/Customer.cs

```diff
@@ -8,16 +8,33 @@ class Customer : Observer
     {
         private readonly string _name;
         private readonly string _favoriteStore;
-        
+
+
         public Customer(string name, string favoriteStore)
         {
             _name = name;
             _favoriteStore = favoriteStore;
+           
 
         }
         public override void Update(float discount)
         {
-            
+            Console.WriteLine($"{_name} Discount is {discount}");
+        }
+
+        public Store AddCustomerToStore(Customer cust, Store store)
```

Comment:
Good/Bad here with these functions

Whats good:
- You have functions to take a customer/store and then handle addObserver. Its abstracted into a simple function.

Whats bad:
- The customer class should not handle adding itself to the store. The observable (customer) shouldnt be telling the store that it wants to be observed.

Imagine in a real life scenario. If Walmart wanted to observe someone, it wouldn't expect the customer to tell Walmart to "watch" them. The company would instead have something setup to detect when someone exits/enters (maybe phone/gps?) but the whole point is.. its an external process to the observer/observable.

With all that said.. for now, given the scope of your problem, I'd move these `AddCustomerToStore` and `RemoveCustomerFromStore` to your `Program.cs`.